File: cotta-guice/src/main/java/com/mgtriffid/games/cotta/experimental/guice/serialization/maps/recipe/MapsChangedEntityRecipe.java
package com.mgtriffid.games.cotta.experimental.guice.serialization.maps.recipe;

import com.mgtriffid.games.cotta.experimental.guice.data.ComponentKey;
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ChangedEntityRecipe;
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ComponentDeltaRecipe;
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ComponentRecipe;

import java.util.List;

public class MapsChangedEntityRecipe implements ChangedEntityRecipe {
    private final int entityId;
    private final List<MapComponentDeltaRecipe> changedComponents;

    public MapsChangedEntityRecipe(int entityId, List<MapComponentDeltaRecipe> changedComponents) {
        this.entityId = entityId;
        this.changedComponents = changedComponents;
    }

    @Override
    public int getEntityId() {
        return entityId;
    }

    @Override
    public List<ComponentDeltaRecipe> getChangedComponents() {
        throw new RuntimeException("TODO");
    }

    @Override
    public List<ComponentRecipe> getAddedComponents() {
        return null;
    }

    @Override
    public List<ComponentKey> getRemovedComponents() {
        return null;
    }
}

File: cotta-guice/src/main/kotlin/com/mgtriffid/games/cotta/experimental/guice/serialization/impl/maps/recipe/MapsChangedEntityRecipe.kt
package com.mgtriffid.games.cotta.experimental.guice.serialization.impl.maps.recipe

import com.mgtriffid.games.cotta.experimental.guice.data.ComponentKey
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ChangedEntityRecipe
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ComponentDeltaRecipe
import com.mgtriffid.games.cotta.experimental.guice.serialization.recipe.ComponentRecipe


class MapsChangedEntityRecipe(
    private val entityId: Int,
    private val changedComponents: List<MapComponentDeltaRecipe>,
    private val addedComponents: List<MapComponentRecipe>,
    private val removedComponents: List<ComponentKey.StringComponentKey>
) : ChangedEntityRecipe {
    override fun getEntityId(): Int {
        TODO("Not yet implemented")
    }

    override fun getChangedComponents(): MutableList<ComponentDeltaRecipe> {
        TODO("Not yet implemented")
    }

    override fun getAddedComponents(): MutableList<ComponentRecipe> {
        TODO("Not yet implemented")
    }

    override fun getRemovedComponents(): MutableList<ComponentKey> {
        TODO("Not yet implemented")
    }
}
